### PR TITLE
feat: Add support for POWHEG weights

### PR DESCRIPTION
**Pull request by APN-Pucky**
powheg uses plain text weights in lhe, like: 

```
<event>
      7  10001  1.11453E+07  2.42132E+00 -1.00000E+00  3.21830E-01
      -2    -1     0     0     0   502  0.000000000E+00  0.000000000E+00  1.596673772E-01  1.596673772E-01  0.000000000E+00  0.00000E+00  9.000E+00
      21    -1     0     0   502   501  0.000000000E+00  0.000000000E+00 -1.979087573E+01  1.979087573E+01  0.000000000E+00  0.00000E+00  9.000E+00
      23     2     1     2     0     0 -2.715131942E-01 -6.004246500E-01 -3.272239123E+00  3.504354871E+00  1.067110589E+00  0.00000E+00  9.000E+00
      11     1     3     3     0     0 -9.921110112E-02  2.574546034E-01 -3.072708859E-01  4.129665530E-01  5.110000000E-04  0.00000E+00  9.000E+00
     -11     1     3     3     0     0 -1.723020930E-01 -8.578792534E-01 -2.964968237E+00  3.091388318E+00  5.110000000E-04  0.00000E+00  9.000E+00
      -2     1     1     2     0   511 -3.204270527E-01  4.049849988E-01 -1.767181124E+00  1.841090837E+00  0.000000000E+00  0.00000E+00  9.000E+00
      21     1     1     2   511   501  5.919402469E-01  1.954396512E-01 -1.459178811E+01  1.460509740E+01  1.685873940E-07  0.00000E+00  9.000E+00
#rwgt            2         101   3933017.7360749426                3         267           0
<weights>
0.11148E+08
0.11098E+08
0.10961E+08
0.13534E+08
0.87849E+07
0.91411E+07
0.13910E+08
0.11218E+08
0.11076E+08
0.10925E+08
0.10971E+08
0.10997E+08
0.11103E+08
0.10964E+08
0.10974E+08
0.11205E+08
0.11144E+08
0.11317E+08
0.11325E+08
0.11321E+08
0.11305E+08
0.11320E+08
0.11363E+08
0.11194E+08
0.11332E+08
0.11284E+08
0.11069E+08
0.11289E+08
0.11223E+08
0.11280E+08
0.11332E+08
0.11131E+08
0.11219E+08
0.11152E+08
0.11150E+08
0.11264E+08
0.11044E+08
0.10938E+08
0.11302E+08
0.11118E+08
0.11175E+08
0.11155E+08
0.11140E+08
0.11353E+08
0.10951E+08
</weights>
</event>
```
[pwgevents-0001.lhe.zip](https://github.com/scikit-hep/pylhe/files/13674230/pwgevents-0001.lhe.zip)

```
* Add support for reading POWHEG weights.
* Add tests for POWHEG.
   - Update scikit-hep-testdata test dependency to include
     version with POWHEG file.
```

### CODE REVIEW

**Comment by eduardo-rodrigues** [comment]
Hello @APN-Pucky, Thank you for this addition 👍. I think it would be very important to add your test file to scikit-hep-testdata and have it used in the test suite here, to enhance what is covered. Can you kindly take care of that?

Agree and any comments, @matthewfeickert?

**Comment by APN-Pucky** [comment]
So just PR a simple LHE file to https://github.com/scikit-hep/scikit-hep-testdata and wait for a release then included that here?

**Comment by eduardo-rodrigues** [comment]
Precisely :)

I will not be much available until Wednesday but you can ping Jim or Matthew for example. Advance thanks.

File: src/pylhe/__init__.py
Comment (APN-Pucky) [inline src/pylhe/__init__.py:329]: One thing I'd like to improve here is to use the label provided in the `initrwgt` block for given index of a weight. But that does not get loaded by default right. Should the `lhe_init` be called here too to translate from index to label.

Comment (APN-Pucky) [reply src/pylhe/__init__.py:329]: eg. for this file it has this sort of init

```
<initrwgt>
<weightgroup name='nominal' combine='None' >
<weight id='MUR1.0_MUF1.0_PDF99208082' >default</weight>
</weightgroup>
<weightgroup name='scale_variations' combine='None' >
<weight id='MUR2.0_MUF2.0_PDF99208082' > renscfact=2d0 facscfact=2d0 </weight>
```
so instead of `weights = {0: XYZ,...}` I'd prefer `weights={'MUR2.0_MUF2.0_PDF99208082': XYZ,...}`
What do you think?

**Comment by eduardo-rodrigues** [comment]
FYI I've now made a release with your files - https://github.com/scikit-hep/scikit-hep-testdata/releases/tag/v0.4.36. I hope it will show up in PyPI soon.

I'm not sure if @matthewfeickert is still around for a review. I will not be working much more until the new year :-).

Enjoy the Christmas break and "talk to you" in 2024!

**Comment by APN-Pucky** [comment]
Thank you for the effort of getting this through in this year. It is by no means in a hurry.

Merry Christmas and happy new year to everyone reading this.

**Comment by APN-Pucky** [comment]
Thanks for adding me to the particle contributors. What's the status of this MR?

**Comment by eduardo-rodrigues** [comment]
Hi. Seems @matthewfeickert and/or @lukasheinrich as experts have been overlooking this. Let's see now with a reping ...

As for your question above, it sounds good to me.

Is there anything else you intend to change in this PR or else it is totally ready from your side?

**Comment by APN-Pucky** [comment]
> Is there anything else you intend to change in this PR or else it is totally ready from your side?

This PR is done from my side since I already changed the access to the weighs to be by the id name.

File: src/pylhe/__init__.py
Comment (eduardo-rodrigues) [inline src/pylhe/__init__.py:310]: Can you improve the docstring to state the cases where this mapping is relevant (I see from the caller below) and state what the argument is and what variables are returned? Not everyone is expert in POWHEG ;-).

File: tests/test_lhe_reader.py
Comment (eduardo-rodrigues) [inline tests/test_lhe_reader.py:149]:
```suggestion
    Test method read_lhe() on several types of LesHouchesEvents POWHEG files.
```

File: tests/test_lhe_reader.py
Comment (eduardo-rodrigues) [inline tests/test_lhe_reader.py:161]:
```suggestion
    Test method read_lhe_with_attributes() on several types of LesHouchesEvents POWHEG files.
```

**Review by eduardo-rodrigues — APPROVED** [review]
Thank you for the nice additions. I'm approving and will be merging tomorrow unless we hear from @matthewfeickert on other feedback.

File: setup.py
Comment (matthewfeickert) [inline setup.py:9]: Are there files in `scikit-hep-testdata` `v0.4.36` that are required here that aren't in previous versions?

File: src/pylhe/__init__.py
Comment (matthewfeickert) [inline src/pylhe/__init__.py:274]: Is this something that is going to be addressed in this PR? If not, then this type of comment would be better served in a GitHub Issue where it can become actionalbe.

File: src/pylhe/__init__.py
Comment (matthewfeickert) [inline src/pylhe/__init__.py:326]:
```suggestion
```

Commented out code shouldn't get checked in.

**Review by matthewfeickert — CHANGES_REQUESTED** [review]
Thanks for the PR @APN-Pucky! This is great, and we should cut a new relase after this. I just have a few questions and requested changes, but this is looking good overall!

File: setup.py
Comment (matthewfeickert) [reply setup.py:9]: Ah, yes indeed looking at https://github.com/scikit-hep/scikit-hep-testdata/releases/tag/v0.4.36. Never mind.

**Review by matthewfeickert — APPROVED** [review]
LGTM. Thank you @APN-Pucky!

**Comment by matthewfeickert** [comment]
@all-contributors please add @APN-Pucky for code.

**Comment by matthewfeickert** [comment]
@APN-Pucky Release [`v0.8.0` is now out on PyPI](https://pypi.org/project/pylhe/0.8.0/) with this contribution in it. Thanks!

**Pull request merged**
